satellite/console/consoleweb/server.go

```diff
@@ -192,7 +212,282 @@ func (server *Server) appHandler(w http.ResponseWriter, r *http.Request) {
+	}
+}
+
+// changeAccountPasswordRequestHandler updates password for a given user.
+func (server *Server) changeAccountPasswordRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type changePasswordRequestModel struct {
+		Password    string `json:"password"`
+		NewPassword string `json:"newPassword"`
+	}
+
+	var passwordChange changePasswordRequestModel
+	err = json.NewDecoder(r.Body).Decode(&passwordChange)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+		return
+	}
+
+	err = server.service.ChangePassword(ctx, passwordChange.Password, passwordChange.NewPassword)
+	if err != nil {
+		server.serveJSONError(w, 404, err)
+		return
+	}
+
+	w.WriteHeader(200)
+}
+
+// createNewUserRequestHandler gets password hash value and creates new inactive User.
+func (server *Server) createNewUserRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type createUserRequestModel struct {
+		console.CreateUser
+		Secret         string `json:"secret"`
+		ReferrerUserID string `json:"referrerUserId"`
+	}
+
+	var model createUserRequestModel
+	err = json.NewDecoder(r.Body).Decode(&model)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+		return
+	}
+
+	secret, err := console.RegistrationSecretFromBase64(model.Secret)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+		return
+	}
+
+	user, err := server.service.CreateUser(ctx, model.CreateUser, secret, model.ReferrerUserID)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+		return
+	}
+
+	token, err := server.service.GenerateActivationToken(ctx, user.ID, user.Email)
+	if err != nil {
+		server.serveJSONError(w, 500, err)
+		return
+	}
+
+	link := server.config.ExternalAddress + ActivationPath + token
+	userName := user.ShortName
+	if user.ShortName == "" {
+		userName = user.FullName
+	}
+
+	server.mailService.SendRenderedAsync(
+		ctx,
+		[]post.Address{{Address: user.Email, Name: userName}},
+		&consoleql.AccountActivationEmail{
+			Origin:         server.config.ExternalAddress,
+			ActivationLink: link,
+		},
+	)
+
+	err = json.NewEncoder(w).Encode(user)
+	if err != nil {
+		server.log.Error(Error.Wrap(err).Error())
+	}
+}
+
+// deleteAccountRequestHandler deletes User.
+func (server *Server) deleteAccountRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	auth, err := console.GetAuth(ctx)
+	if err != nil {
+		server.serveJSONError(w, 401, err)
+		return
+	}
+
+	type deleteAccountRequestModel struct {
+		Password string `json:"password"`
+	}
+
+	var password deleteAccountRequestModel
+	err = json.NewDecoder(r.Body).Decode(&password)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+		return
+	}
+
+	err = server.service.DeleteAccount(ctx, password.Password)
+	if err != nil {
+		server.serveJSONError(w, 404, err)
```

Comment:
Here the error could be a `500`.

About `404`, I feel that it isn't appropriated, I feel more to be a `422`, but I'm not feeling very strong on this.